### ISSUE TITLE
Copying/moving/linking input/output/restart data?

### ISSUE THREAD

**Issue by joakimkjellsson** [esm-tools/esm_tools#163]
Hi all

I have an experiment with OpenIFS running currently on HLRN. 
I've noticed that the model execution is ~1h40m, but the whole job can easily take 2h. I think the extra time is copying input data to the work dir and copying output/restart to outdata/restart dirs. 
I have checked the compute node towards the end, and it is only one task "esm_runscripts" running, and only using 20-30% CPU. 
Since this only runs as a single task it is a waste of computer resources. 

So my questions/feature requests are two: 
1) Can we replace some of the copying by linking? Input/forcing data only needs to be linked to the work dir rather than copied. For example, input data for high-res NEMO is insanely big and it does not make sense to make multiple copies of this. FESOM mesh files are another example. 
2) Things that need to be copied, e.g. model output, could be done by another job running on only one task. 
Once this is done, we could remove some data from the work dir to avoid amassing duplicates of output data. 

Cheers
Joakim

**Comment by pgierz** [comment]
Hi Joakmin,

Good idea; properly implementing this will need the following, I think:

- Re-write of the part that copies around files (this is messy and could use a cleanup and docu anyway!)
- Some way to determine if the files you need (in your case, for input files, but this should work for whatever filetype) are links or copies.

Probably the user would then specify something like:

```yaml

open_ifs:
    input_links:
        file_tag1: file_one
    input_links_sources:
         file_tag1: /some/path/on/hlrn
     input_links_in_work:
         file_tag1: name_model_wants.nc

     input_files:
          file_tag2: this_file_is_copied_and_not_linked
     input_sources:
          file_tag2: /some/path
     input_in_work:
           file_tag2: other_name.nc
```

What do you think?

**Comment by joakimkjellsson** [comment]
Sounds like a good idea to me! 
Cheers
Joakim

**Comment by seb-wahl** [comment]
I think this should have high priority as output/restarts/input will be even larger once we add AGRIF operationally. So I suggest the workflow may be as follows (this is roughly how we do it in our old runtime environment based on MPI's mkexp tool):
1. Link all input files to work instead of copying
2. run the simulation
3. once simulation has finished, copy/move restarts in place (can't be done by postprocessing in 5. because they need to be in place before we resubmit in 4.)
4. resubmit run 
5. At the same time resubmit post/store job that cleans the work directory, and triggers the "real" postprocessing e.g. model2pressure level conversion, whatever...

I tried a while ago to have a go at this myself, but miserably failed because I got completely lost in that part of the code.
Nevertheless, I'm happy to help.
